tests/agents/microchain/conftest.py

```diff
@@ -23,25 +23,63 @@ def mock_summaries(function_names: list[str]) -> Summaries:
 
 
 @pytest.fixture(scope="module")
-def wxdai_contract_mocked_rag() -> Generator[ContractClassConverter, None, None]:
-    with patch(
-        "prediction_market_agent.agents.microchain_agent.blockchain.code_interpreter.CodeInterpreter.generate_summary",
-        Mock(side_effect=mock_summaries),
-    ), patch(
-        "prediction_market_agent.agents.microchain_agent.blockchain.code_interpreter.CodeInterpreter.build_retriever",
-        Mock(return_value=None),
-    ), patch(
-        "prediction_market_agent.agents.microchain_agent.blockchain.code_interpreter.CodeInterpreter.build_rag_chain",
-        Mock(return_value=None),
-    ):
-        wxdai = WrappedxDaiContract()
-        contract_address = Web3.to_checksum_address(wxdai.address)
-        c = ContractClassConverter(contract_address)
-        yield c
+def wxdai_contract_class_converter() -> Generator[ContractClassConverter, None, None]:
+    wxdai = WrappedxDaiContract()
+    contract_address = Web3.to_checksum_address(wxdai.address)
+    yield ContractClassConverter(
+        contract_address=contract_address, contract_name=wxdai.__class__.__name__
+    )
+
+
+class RAGPatcherManager:
```

Comment:
Do we need this anymore now you've removed the RAG stuff from `build_functions_from_smart_contract`?

Reply:
Just for testing, we still need that - so that we skip the LLM part of things for not spending OpenAI credits. But I will rename it.